managedplugin/download.go

```diff
@@ -92,7 +92,7 @@ func getURLLocation(ctx context.Context, org string, name string, version string
 	return "", fmt.Errorf("failed to find plugin %s/%s version %s", org, name, version)
 }
 
-func DownloadPluginFromHub(ctx context.Context, authToken, localPath, team, name, version string, typ PluginType) error {
+func DownloadPluginFromHub(ctx context.Context, authToken, teamName, localPath, pluginTeam, name, version string, typ PluginType) error {
```

Comment:
I wonder if it's time to use a struct here so it's not so easy to get the order of all the string parameters wrong

Reply:
I would love that but how far to take it? This pattern is everywhere and technically the methods are public.

Reply:
Technically public yes, but we're already breaking the signature with this change by adding a parameter, right? May as well make it a struct

Maybe something like `func DownloadPluginFromHub(ctx context.Context, opts HubDownloadOptions)` with 

```
type HubDownloadOptions struct {
    AuthToken string
    TeamName string
    LocalPath string
    PluginTeam string
    PluginKind string
    PluginName string
    PluginVersion string
}
```

Reply:
Good suggestion - added in https://github.com/cloudquery/plugin-pb-go/pull/162/commits/2bdb2218412f68e9432ab57d100b5b1506130739